### PR TITLE
Babel task: make the app to work in IE10, add babel plugin to remove console.log expressions

### PR DESCRIPTION
**Pull request by YuryVegero**
In scope of the PR I did the following:
1. added babel to the project
2. added polyfills
3. added browserify and babelify to bundle js code
4. refactored code to work in IE10 (changes connected with event handlers)
5. wrote babel plugin to remove  from the code all console.log expressions.

### CODE REVIEW

File: news-portal/src/js/modules/dom/utils/dom.js [hunk 1]
```diff
@@ -26,14 +26,21 @@ export const findAll = (selector, context = document) => [...context.querySelect
...
     if (isIterable(elements)) {
-        elements.forEach(elem => elem.addEventListener(type, handler)); // example of 'forEach' loop usage
+        elements.forEach(elem => bindEvent(elem, ...rest)); // example of 'forEach' loop and '...rest'
     } else {
-        elements.addEventListener(type, handler);
+        bindEvent(elements, ...rest);
     }
 };
 
+function bindEvent(el, type, handler) {
```
[the remark this answers is not in the document]
Comment (YuryVegero) [reply news-portal/src/js/modules/dom/utils/dom.js]: changed to use arrow function

File: news-portal/src/js/modules/dom/utils/dom.js [hunk 2]
```diff
@@ -26,14 +26,21 @@ export const findAll = (selector, context = document) => [...context.querySelect
...
+    if (el.attachEvent) {
+        el.attachEvent('on' + type, handler);
```
[the remark this answers is not in the document]
Comment (YuryVegero) [reply news-portal/src/js/modules/dom/utils/dom.js]: changed to use template

File: news-portal/src/js/modules/dom/renderers/channels.js [hunk 3]
```diff
@@ -23,4 +27,32 @@ export const switchActiveChannel = (activeChannel, channelElements) => {
...
     dom.removeClass(prevActiveChannel, activeChannelClass);
     dom.addClass(activeChannel, activeChannelClass);
+};
+
+export const loadAndRenderChannels = (container) => {
+    return api.fetchChannels().then(channels => renderChannels(channels, container));
+};
+
+export const bindChannelClick = (loadArticlesCallback) => {
+    const channelsContainer = dom.findOne('.channels__body');
```
[the remark this answers is not in the document]
Comment (YuryVegero) [reply news-portal/src/js/modules/dom/renderers/channels.js]: fixed

File: news-portal/src/js/modules/dom/renderers/channels.js [hunk 4]
```diff
@@ -23,4 +27,32 @@ export const switchActiveChannel = (activeChannel, channelElements) => {
     });
     dom.removeClass(prevActiveChannel, activeChannelClass);
     dom.addClass(activeChannel, activeChannelClass);
+};
+
+export const loadAndRenderChannels = (container) => {
+    return api.fetchChannels().then(channels => renderChannels(channels, container));
+};
+
+export const bindChannelClick = (loadArticlesCallback) => {
```
[the remark this answers is not in the document]
Comment (YuryVegero) [reply news-portal/src/js/modules/dom/renderers/channels.js]: refactored to use classes

File: news-portal/package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "babel-cli": "^6.26.0",
+    "babel-core": "^6.26.0",
+    "babel-preset-env": "^1.6.1",
+    "babelify": "^8.0.0",
+    "browserify": "^14.5.0"
+  },
+  "dependencies": {
+    "babel-polyfill": "^6.26.0",
+    "es6-promise": "^4.1.1",
+    "isomorphic-fetch": "^2.2.1"
```
[the remark this answers is not in the document]
Comment (YuryVegero) [reply news-portal/package.json]: refactored to use XHR wrapped with generator

File: news-portal/src/js/app.js [hunk 6]
```diff
@@ -9,24 +10,16 @@ const {
     ]
 } = dom.findOne('main'); // example of object and array destructuring
```
Comment (yury-kuzmiankov) [inline news-portal/src/js/app.js]: Good to move strings to constants

File: news-portal/src/js/app.js [hunk 7]
```diff
@@ -9,24 +10,16 @@ const {
...
+    container: channelsContainer,
+    handleClick: (channel) => {
+        new Articles(channel, articlesContainer).init();
+    }
+};
 
+const channelsCmp = new Channels(channelsConfig);
+channelsCmp.init();
 
+console.log("One more test log output which will not be shown!");
```
Comment (yury-kuzmiankov) [inline news-portal/src/js/app.js]: Is it required for app?

Comment (YuryVegero) [reply news-portal/src/js/app.js]: Yes, this is to test that my custom babel plugin works properly to remove all console.log expressions from the final bundle. 
Here is the path to the plugin's code: src/js/plugins/babel-plugin-remove-log/index.js

**Pull request merged**